SpatialGDK/Source/SpatialGDKEditor/Private/SchemaGenerator/SpatialGDKEditorSchemaGenerator.cpp

```diff
@@ -494,15 +857,36 @@ bool SaveSchemaDatabase(const FString& PackagePath)
 	SchemaDatabase->LevelComponentIds.Reset(LevelPathToComponentId.Num());
 	LevelPathToComponentId.GenerateValueArray(SchemaDatabase->LevelComponentIds);
 
+	SchemaDatabase->ComponentSetIdToComponentIds.Reset();
+	for (const auto& WellKnownComponent : SpatialConstants::ServerAuthorityWellKnownComponents)
+	{
+		SchemaDatabase->ComponentSetIdToComponentIds.FindOrAdd(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID)
+			.ComponentIDs.Push(WellKnownComponent.Key);
+	}
+	for (const auto& WellKnownComponent : SpatialConstants::ClientAuthorityWellKnownComponents)
+	{
+		SchemaDatabase->ComponentSetIdToComponentIds.FindOrAdd(SpatialConstants::CLIENT_AUTH_COMPONENT_SET_ID)
+			.ComponentIDs.Push(WellKnownComponent.Key);
+	}
+	SchemaDatabase->ComponentSetIdToComponentIds.FindOrAdd(SpatialConstants::GDK_KNOWN_ENTITY_AUTH_COMPONENT_SET_ID)
+		.ComponentIDs.Append(SpatialConstants::KnownEntityAuthorityComponents);
+	SchemaDatabase->ComponentSetIdToComponentIds.FindOrAdd(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID)
+		.ComponentIDs.Append(NetCullDistanceComponentIds);
```

Comment:
do we add/remove NCD to change their NCDs at runtime?

Reply:
We don't currently - but have write auth over them all the same - previous comment in the entity factory was -
`// Add all Interest component IDs to allow us to change it if needed.`
Seems like it _might_ be useful someday, and not sure it hurts to have it. Happy to remove them if you prefer though.